courses/free-courses-hi.md

```diff
@@ -213,6 +213,7 @@
 
 * [Compiler Design](https://www.youtube.com/playlist?list=PL9FuOtXibFjVR-87LcU-DS-9TJcbG97_p) - Abhishek Sharma
 * [Compiler Design](https://www.youtube.com/playlist?list=PL23dd-8zssJBiyntds3X1sVWUDeb0Aa1N) - Start Practicing
+* [Compiler Design](https://www.youtube.com/playlist?list=PLmXKhU9FNesSmu-_DKC7APRoFkaQvGurx) - KnowledgeGate
```

Comment:
Don't invent titles! Use the title given by the resource.
Please add creator names/channel name where appropriate.